### ISSUE TITLE
Provide std::error::Error impl for SignatureError

### ISSUE THREAD

**Issue by dtolnay** [dalek-cryptography/ed25519-dalek#103]
ed25519_dalek::SignatureError does not implement std::error::Error, which makes it unergonomic to use with modern error libraries that rely on that impl for error types.

```rust
use anyhow::Result;
use ed25519_dalek::SignatureError;

fn f() -> Result<(), SignatureError> {
    Ok(())
}

fn main() -> Result<()> {
    f()?;  // doesn't work
    Ok(())
}
```

```console
error[E0277]: the trait bound `ed25519_dalek::errors::SignatureError: std::error::Error` is not satisfied
 --> src/main.rs:9:8
  |
9 |     f()?;  // doesn't work
  |        ^ the trait `std::error::Error` is not implemented for `ed25519_dalek::errors::SignatureError`
  |
  = note: required because of the requirements on the impl of `std::convert::From<ed25519_dalek::errors::SignatureError>` for `anyhow::Error`
  = note: required by `std::convert::From::from`
```

**Comment by tarcieri** [comment]
Note that #102 would address this

**Comment by dtolnay** [comment]
Ah nice, thanks!

**Comment by isislovecruft** [comment]
Hi! Thanks for the suggestion. We'd be happy to provide an impl of `std::error::Error`.

**Comment by isislovecruft** [comment]
Hi! This was fixed in #134 and will be available in 1.0.0-pre.4 which I am about to release.
